Fix Round order prioritization and boosting logic

andrebastosdias opened the pull request:
Tested it on the console

Edit: https://www.smogon.com/forums/threads/instruct-should-boost-round.3764935/
Round is boosted if it was used prior in the turn, even if called by Instruct or Encore.

andrebastosdias committed:
Fix move order prioritization

andrebastosdias committed:
Fix test

andrebastosdias committed:
A crit in the test

andrebastosdias committed:
Modify test

andrebastosdias committed:
Merge branch 'master' into round

andrebastosdias committed:
Update to ESLint 9

DaWoblefet commented on test/sim/moves/round.js:
```diff
@@ -18,6 +18,23 @@ describe('Round', () => {
+			{ species: "Blastoise", moves: ['sleeptalk'] },
+			{ species: "Incineroar", ability: 'intimidate', moves: ['sleeptalk'] },
+		]]);
+		battle.makeChoices();
```
You should be explicit who you are targeting here with Round.

andrebastosdias committed:
Make Round targets explicit

andrebastosdias committed:
Update action orders

andrebastosdias committed:
Merge branch 'smogon:master' into round

andrebastosdias committed:
Merge branch 'smogon:master' into round

andrebastosdias committed:
Fix Round boost logic

andrebastosdias commented:
https://www.smogon.com/forums/threads/instruct-should-boost-round.3764935/
Round is boosted if it was used prior in the turn, even if called by Instruct or Encore.

andrebastosdias committed:
Fix test name

andrebastosdias committed:
Merge branch 'smogon:master' into round

andrebastosdias committed:
Merge branch 'smogon:master' into round

andrebastosdias committed:
Fix order

andrebastosdias committed:
Change roundBoost type

andrebastosdias committed:
Merge branch 'round' of https://github.com/andrebastosdias/pokemon-showdown into round

andrebastosdias committed:
Merge remote-tracking branch 'upstream/master' into round

andrebastosdias committed:
Simplify to a Round counter

andrebastosdias committed:
Boost Round even if there is not target

https://www.smogon.com/forums/threads/instruct-should-boost-round.3764935/post-10645271

andrebastosdias committed:
Implement like Echoed Voice

andrebastosdias committed:
Merge branch 'master' into round

andrebastosdias committed:
Merge branch 'master' into round

andrebastosdias committed:
Merge branch 'master' into round

andrebastosdias committed:
The condition doesn't need a Start event

andrebastosdias committed:
Merge branch 'master' into round
